resourceGroup.tf

```diff
@@ -1,3 +1,4 @@
+#Terraform settings block
 terraform {
     required_version = ">= 1.0.0"
     required_providers {
```

Comment:
Move the providers and required providers into its own file - call it either providers.tf or common.tf; so it's kept away.

Also use version 3.0 of azurerm, it's a BIG improvement, worth reading up on it. 
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide